### PR TITLE
LGA-1772 - Update Django Rest Framework(DRF) to 2.4.8

### PR DESCRIPTION
**Pull request by said-moj**
## What does this pull request do?

- Update Django Rest Framework(DRF) to 2.4.8
- Update DRF Nested Routes package to 0.9.0
- Remove usage of deprecated `@action` and `@link` decorators in favour of recommended `@detail_route`

## Any other changes that would benefit highlighting?
[2.4 Announcement ](http://www.tomchristie.com/rest-framework-2-docs/topics/2.4-announcement)
[2.4 release notes](https://github.com/encode/django-rest-framework/blob/version-2.4.x/docs/topics/release-notes.md#240)

The update to DRF Nested Routes was required because the previous version [0.1.3 stopped working with DRF 2.4](https://github.com/alanjds/drf-nested-routers/issues/24)

rest_framework.compat no longer exists in DRF 2.4  - [rest_framework.compat.url was imported from django.conf.urls.url](https://github.com/encode/django-rest-framework/blob/2.3.14/rest_framework/compat.py#L22), so we directly import from there now

Removed core.drf.routers.Route, core.drf.routers. DynamicDetailRoute and core.drf.routers.DynamicListRoute - These are now available as part of the DRF in `from rest_framework.routers import Route, DynamicDetailRoute, DynamicListRoute`

Used DynamicDetailRoute instead of Route for dynamically generated links so that [at]action and [at]link routes are added as routes

There was a [change in the way DRF does field_mapping in 2.4](https://github.com/encode/django-rest-framework/pull/1818) which resulted in the `legalaid.fields.MoneyField` being mapped to a `rest_framework.fields.IntegerField` and the validation for min and max values not working. To work around this issue we introduced a new `legalaid.fields.MoneyFieldDRF` field that is added to the field_mapping so that MoneyField min and max validators run.

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: cla_backend/apps/call_centre/views.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 from legalaid.permissions import IsManagerOrMePermission
 
 from rest_framework import viewsets, mixins, status
-from rest_framework.decorators import action, link
+from rest_framework.decorators import detail_route, list_route
```
Comment (jogilder) [inline cla_backend/apps/call_centre/views.py:18]: can we remove @list_route as we don't use it?

File: cla_backend/apps/core/drf/router.py [hunk 2]
```diff
@@ -1,21 +1,15 @@
...
-
 from rest_framework import views
 from rest_framework.routers import BaseRouter, flatten, replace_methodname
 from rest_framework.urlpatterns import format_suffix_patterns
-from rest_framework.compat import url
+from django.conf.urls import url
 from rest_framework.reverse import reverse
 from rest_framework.response import Response
 
 from rest_framework_nested.routers import NestedSimpleRouter as OriginalNestedSimpleRouter
```
Comment (jogilder) [inline cla_backend/apps/core/drf/router.py:11]: did you look to see if the parts in this file that are `Borrowed directly from DRF v3.0.2.` could be removed?

File: cla_backend/apps/call_centre/views.py (hunk 1 above)
Comment (said-moj) [reply cla_backend/apps/call_centre/views.py:18]: It's used lowered down in this file https://github.com/ministryofjustice/cla_backend/blob/feature/LGA-1772-drf-2.4.8/cla_backend/apps/call_centre/views.py#L265

This was actually already a `@list_route` which was being imported from `from core.drf.decorators import list_route`

**Review by Pheonnexx — APPROVED** [review]

**Review by psweeting1 — APPROVED** [review]
LGTM

File: cla_backend/apps/core/drf/router.py (hunk 2 above)
Comment (jogilder) [reply cla_backend/apps/core/drf/router.py:11]: Understood

**Pull request merged**